### DISCUSSION TITLE
getResolvedFileUri not defined?

### DISCUSSION THREAD

**Discussion by hsd-dev** [eclipse-emfcloud/emfcloud#189] [ModelServer] [answered]
I am trying to implement a model server for my graphical editor. Since my model is based on Ecore, I am using [org.eclipse.emfcloud.ecore.modelserver](https://github.com/eclipse-emfcloud/ecore-glsp/tree/master/server/org.eclipse.emfcloud.ecore.modelserver) as a reference. In the [EcoreModelServerRouting](https://github.com/eclipse-emfcloud/ecore-glsp/blob/master/server/org.eclipse.emfcloud.ecore.modelserver/src/main/java/org/eclipse/emfcloud/ecore/modelserver/EcoreModelServerRouting.java), I could not find references to [getResolvedFileUri](https://github.com/eclipse-emfcloud/ecore-glsp/blob/master/server/org.eclipse.emfcloud.ecore.modelserver/src/main/java/org/eclipse/emfcloud/ecore/modelserver/EcoreModelServerRouting.java#L52) function. Where has it been defined?

**Comment by ndoschek** [comment, marked as answer]
Hi @ipa-hsd,

Please note that the `@eclipse-emfcloud/ecore-glsp` repository contains a demonstrator for showing an editor for Ecore models. Thus, we only focus on what’s relevant for the demonstration under a controlled environment.

That means, the development state of this project is outdated (i.e. it uses ModelServer V1).
This is also the reason why you cannot find references to the linked method, as the URI handling was re-worked and improved with ModelServer V2.
I strongly recommend to use an updates project as reference, e.g. [`org.eclipse.emfcloud.coffee.modelserver`](https://github.com/eclipse-emfcloud/coffee-editor/tree/master/backend/plugins/org.eclipse.emfcloud.coffee.modelserver), as this was recently updated to the latest versions.

PS: If you are interested in transforming this demonstrator into an adoptable state, e.g. by funding the development work or working on it yourself, please let us know!
